### PR TITLE
Auth fixes

### PR DESCRIPTION
**Pull request by mikesch803**
1. add the logout handler
2. userstate will remain after the refresh
3. add 404 page
4. add form validation
5. add categories api integration on landing page, and changed the landing banner

email: adarshbalika@gmail.com password : adarshbalika

## [live link](https://624b1bb5a3807020d3e0ea11--agitated-tesla-b2afa9.netlify.app/)

### CODE REVIEW

File: src/context/wishlist-context.jsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ const WishlistProvider = ({ children }) => {
   const encodedToken = localStorage.getItem("token");
   const {setToastMsg, setToastState } =
     useContext(ToastContext);
-
+console.log('from context wishlist ' ,wishlistData)
```
Comment (nayyyhaa) [inline src/context/wishlist-context.jsx:12]: You can remove the console log if not required anymore.

File: src/pages/signup-page/Signup.jsx [hunk 2]
```diff
@@ -20,70 +20,139 @@ const [state, dispatch] = useReducer(AuthReducer, {
...
-  }
+
+    if (state.field.email.indexOf("@") === -1) {
+      dispatch({ type: "EMAIL_ERR" });
+    } else {
+      dispatch({ type: "EMAIL_ERR" });
+    }
+
+    if (state.field.password.length < 8) {
+      dispatch({ type: "PASSWORD_ERR" });
```
Comment (nayyyhaa) [inline src/pages/signup-page/Signup.jsx:52]: Both if & else condition are dispatching same action. Why is that so?

Comment (mikesch803) [reply src/pages/signup-page/Signup.jsx:52]: so basically it show err, here i m using toggle, so if user correctly input, the same condition turns false and err will be removed

File: src/pages/login-page/Login.jsx [hunk 3]
```diff
@@ -9,42 +9,39 @@ import { ToastContext } from "../../context/toast-context";
...
+        if (response.status === 200) {
+          localStorage.setItem("token", response.data.encodedToken);
+          setLogin(true);
+          setToastState(true);
+          setToastMsg("Signin sucessfully");
+          setTimeout(() => {
+            setToastState(false);
+          }, 1500);
+        }
+        console.log(response);
```
Comment (BalajiHarinath) [inline src/pages/login-page/Login.jsx:36]: Suggestion :- Can remove console log if not required

File: src/context/wishlist-context.jsx [hunk 4]
```diff
@@ -27,6 +27,7 @@ const WishlistProvider = ({ children }) => {
         if (response.status === 200) {
           setWishlistData(response.data.wishlist);
           setToastState(true);
+          console.log(response.data.wishlist)
```
Comment (BalajiHarinath) [inline src/context/wishlist-context.jsx:30]: Suggestion :-  Can remove console log here

File: src/pages/landing-page/LandingPage.jsx [hunk 5]
```diff
@@ -1,35 +1,44 @@
...
+    (async () => {
+      try {
+        const response = await axios.get("/api/categories");
+        setBrands(response.data.categories);
+        dispatch({type:"CLEAR"})
+      } catch (err) {
+        console.error(err);
+      }
+    })();
+  }, []);
```
Comment (BalajiHarinath) [inline src/pages/landing-page/LandingPage.jsx:22]: Appreciation :- Good use of IIFE

File: src/pages/signup-page/Signup.jsx [hunk 6]
```diff
@@ -20,70 +20,139 @@ const [state, dispatch] = useReducer(AuthReducer, {
...
+    } else {
+      dispatch({ type: "PASSWORD_ERR" });
+    }
+
+    if (state.field.password !== state.field.confirmPassword) {
+      dispatch({ type: "CONFIRM_PASSWORD_ERR" });
+    } else {
+      dispatch({ type: "CONFIRM_PASSWORD_ERR" });
+    }
```
Comment (BalajiHarinath) [inline src/pages/signup-page/Signup.jsx:61]: Suggestion :- You can have different states for if-else just to improve the readability of the code.

**Review by BalajiHarinath — APPROVED** [review]
Apart from the suggestions everything looks good.

File: src/pages/signup-page/Signup.jsx (hunk 6 above)
Comment (mikesch803) [reply src/pages/signup-page/Signup.jsx:61]: it is toggle to show/hide the err msg, which turns false if you user input correctly

**Pull request merged**